cli/root.go

```diff
@@ -85,15 +83,30 @@ func NewAlertmanagerClient(amURL *url.URL) *client.Alertmanager {
 
 	cr := clientruntime.New(address, path.Join(amURL.Path, defaultAmApiv2path), schemes)
 
-	cr.Transport = &http.Transport{
-		TLSClientConfig: tlsConfig,
+	if amURL.User != nil && httpConfigFile != "" {
+		kingpin.Fatalf("basic authentication and http.config.file are mutually exclusive")
 	}
 
 	if amURL.User != nil {
 		password, _ := amURL.User.Password()
 		cr.DefaultAuthentication = clientruntime.BasicAuth(amURL.User.Username(), password)
 	}
 
+	var httpConfig *promconfig.HTTPClientConfig
```

Comment:
I'd find it more readable if we had distinct paths depending on whether a config file is given or not
```
var cr *clientruntime.Runtime
if httpConfigFile != "" {
   ...
} else {
   ...
}
```

Reply:
remove this variable declaration since we do not need to have this variable out of the if statement.